packages/gatsby-link/src/index.js

```diff
@@ -151,6 +156,9 @@ class GatsbyLink extends React.Component {
               }
             }
 
+            // Temp hack while awaiting https://github.com/reach/router/issues/119
+            window.__navigatingToLink = true
```

Comment:
I think this should be in https://github.com/gatsbyjs/gatsby/blob/5f09cc377d06e51df5b39abc2fb5300853907c6b/packages/gatsby/cache-dir/navigation.js#L50 rather than `Link` component to cover programmatic navigation

Reply:
and yeah, replacing shouldn't update the scroll